modules/api/modules/v1/components/BaseController.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace app\modules\api\modules\v1\components;
+
+use Yii;
+use app\modules\api\modules\v1\models\User;
+use app\modules\api\modules\v1\Module;
+use yii\filters\AccessControl;
+use yii\filters\auth\CompositeAuth;
+use yii\filters\auth\HttpBasicAuth;
+use yii\filters\auth\HttpBearerAuth;
+use yii\filters\auth\QueryParamAuth;
+use yii\rest\Controller;
+use yii\rest\Serializer;
+use yii\web\Request;
+use yii\web\Response;
+
+/**
+ * @property Response $response
+ * @property Request $request
+ * @property User $currentUser
+ * 
+ * @property Module $module
+ */
+class BaseController extends Controller
+{
+    public $serializer = [
+        'class' => Serializer::class,
+        'collectionEnvelope' => 'items',
+    ];
+    
+    /**
+     * @var User|bool
+     */
+    private $_currentUser;
+
+    public function init()
+    {
+        parent::init();
+        
+        $this->response->on(Response::EVENT_BEFORE_SEND, function ($event) {
+            /** @var Response $response */
+            $response = $event->sender;
+            
+            $response->data = [
+                'status' => $response->isSuccessful ? 'ok' : 'error',
+                'data' => $response->data,
+            ];
+            $response->statusCode = 200;
```

Comment:
In case of error status code should not be 200.

Reply:
Then there is no sense in wrapping:
```
{
    "status": "error",
    "data": {
        "name": "Not Found",
        "message": "The requested news does not exist",
        "code": 0,
        "status": 404,
        "type": "yii\\web\\NotFoundHttpException"
    }
}
````

In this case when comes code `5**`, it means the server is **very sick**). To look the `data` is already meaningless.

Reply:
There's no sense giving out 200 code for 404 error or 400 error.